File: BOJ/silver4/boj_10828.py
# 10828. 스택

import sys
input = sys.stdin.readline

n = int(input())
stack = []

for _ in range(n):
    c = input().rstrip().split()
    if c[0] == 'push':
        stack.append(int(c[1]))
        
    elif c[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())

    elif c[0] == 'size':
        print(len(stack))

    elif c[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)

    elif c[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: BOJ/silver4/boj_10828.cpp
// 10828. 스택
#include <bits/stdc++.h>
using namespace std;

int main(void)
{
  ios::sync_with_stdio(0);
  cin.tie(0);

  int n;
  cin >> n;
  stack<int> s;

  while (n--)
  {
    string command;
    cin >> command;
    if (command == "push")
    {
      int data;
      cin >> data;
      s.push(data);
    }
    else if (command == "pop")
    {
      if (s.empty())
        cout << -1 << "\n";
      else
      {
        int x = s.top();
        cout << x << "\n";
        s.pop();
      }
    }
    else if (command == "size")
      cout << s.size() << "\n";
    else if (command == "empty")
    {
      if (s.empty())
        cout << 1 << "\n";
      else
        cout << 0 << "\n";
    }
    else if (command == "top")
      if (s.empty())
        cout << -1 << "\n";
      else
        cout << s.top() << "\n";
  }
}